index.js

```diff
@@ -62,62 +62,49 @@ function createWarning (name, code, message, { unlimited = false } = {}) {
 
   code = code.toUpperCase()
 
-  return new WarningItem(name, code, message, unlimited)
-}
+  const warning = unlimited 
+  ? function (a, b, c) {
+    warning.emitted = true
 
-/**
- * Represents a warning item with details.
- * @class
- * @memberof processWarning
- * @param {string} name - The name of the warning.
- * @param {string} code - The code associated with the warning.
- * @param {string} message - The warning message.
- * @param {boolean} unlimited - If true, allows unlimited emissions of the warning.
- */
-class WarningItem {
-  constructor (name, code, message, unlimited) {
-    this.name = name
-    this.code = code
-    this.message = message
-    this.unlimited = unlimited
-    this.emitted = false
+    process.emitWarning(warning.format(a, b, c), warning.name, warning.code)
   }
-
-  /**
-   * Emits the warning.
-   * @param {*} [a] Possible message interpolation value.
-   * @param {*} [b] Possible message interpolation value.
-   * @param {*} [c] Possible message interpolation value.
-   */
-  emit (a, b, c) {
-    if (this.emitted === true && this.unlimited === false) {
+  : function (a, b, c) {
+    if (warning.emitted === true) {
       return
     }
-    this.emitted = true
-
-    process.emitWarning(this.format(a, b, c), this.name, this.code)
+    warning.emitted = true
+    
+    process.emitWarning(warning.format(a, b, c), warning.name, warning.code)
   }
 
+  warning.emitted = false;
+  warning.message = message;
+  warning.unlimited = unlimited;
+  Object.defineProperty(warning, 'name', { value: name })
```

Comment:
this one slows it down, because it has to lookup the name

Reply:
why is this needed?

Reply:
Because a function has a non writable name.

Reply:
Why do you need to change it

Reply:
Because we accept a `name` parameter as input, that can be read by the user.

@Uzlopak What if we rename it to `group` instead?

Reply:
I think name is important to detect DeprecationWarnings so that node would ignore them. 

The function name is usually set by doing `function NAME () {}` or `const NAME = () => {}`, where `NAME` is the desired Function Name. But I dont know how I could do it at runtime. Maybe by doing abusing `new Function`, which is basically a `eval` call. Also some runtimes disable, i think cloudflare, disallow `new Function` because it is under the hood an eval. 

It could be also done by doing:

```js
const NAME = 'DeprecationWarning'
const container = {
  [NAME]: () => {}
}

console.log(container[NAME].name) // 'DeprecationWarning'
```

But something like 
```js
function createNamedFunction (name, fn = () => {}) {
  const container = {
    [name]: fn
  }
    return container[name]
}

const bla = createNamedFunction('DeprecationWarning', () => {})
console.log(bla.name) //  '' <-- empty
```